### PR TITLE
Add Static Tables to Optimism V2

### PR DESCRIPTION
**Pull request by MSilb7**
Copied the Ethereum formats for erc20 and nft tokens for Optimism.

Also added the Optimism airdrop list (please check this format / alias to see if it makes sense).

Would we also be able to add ERC20/721/1155 transfers to the Optimism V2 database?

I've checked that:

* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] the directory tree matches the pattern /sector/blockchain/ e.g. /tokens/ethereum
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each file has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`

### CODE REVIEW

**Comment by MSilb7** [comment]
cc @soispoke - Attempt 1 at adding info tables for Optimism in V2

**Review by soispoke — COMMENTED** [review]
Thanks for your submission MSilb7 ! Sorry it took a bit longer than usual because of our team offsite

A couple things that need to be added:
- A tokens_optimism_schema.yml file (similar to https://github.com/duneanalytics/abstractions/blob/master/spellbook/models/tokens/ethereum/tokens_ethereum_schema.yml)
- Modify the dbt_project.yml file (just add optimism schema under tokens after line 66: https://github.com/duneanalytics/abstractions/blob/master/spellbook/dbt_project.yml)
- Can you add a standard to tokens_optimism_nft.sql ? (and maybe symbol as well if possible)

**Comment by MSilb7** [comment]
> Thanks for your submission MSilb7 ! Sorry it took a bit longer than usual because of our team offsite
> 
> A couple things that need to be added:
> 
> * A tokens_optimism_schema.yml file (similar to https://github.com/duneanalytics/abstractions/blob/master/spellbook/models/tokens/ethereum/tokens_ethereum_schema.yml)
> * Modify the dbt_project.yml file (just add optimism schema under tokens after line 66: https://github.com/duneanalytics/abstractions/blob/master/spellbook/dbt_project.yml)
> * Can you add a standard to tokens_optimism_nft.sql ? (and maybe symbol as well if possible)

Hey no worries, I'm squeezing this in between other work as well.

Just added schema files, will get on the extra NFT columns next. We should be able to script the data pull given the contract addresses. I'll share the script here once it's done - should be helpful for other migrations (unless one already exists).

**Review by jeff-dude — APPROVED** [review]
this looks good to me, but i'll let @soispoke or @dot2dotseurat give a final approval

File: spellbook/models/tokens/optimism/tokens_optimism_schema.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+version: 1
+
+models:
+  - name: tokens_ethereum_erc20
```
Comment (soispoke) [inline spellbook/models/tokens/optimism/tokens_optimism_schema.yml:4]: should be tokens_optimism_erc20

File: spellbook/models/tokens/optimism/tokens_optimism_schema.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    meta:
+      blockchain: optimism
+      sector: tokens
+      project: erc20
+      contibutors: msilb7
+    config:
+      tags: ['table', 'erc20']
```
Comment (soispoke) [inline spellbook/models/tokens/optimism/tokens_optimism_schema.yml:11]: adding the chain, 'optimism' as a tag is often useful

**Review by soispoke — COMMENTED** [review]
Few minor comments but it looks good 👍

**Comment by dot2dotseurat** [comment]
@MSilb7 there's a couple changes needed, I'm going to add them over here so I can use our dbt cloud tests https://github.com/duneanalytics/abstractions/pull/1306 

1) There were some missing quotes in the optimism token addresses files.
2) Schema files need to be be verison 2. It's misleading but I believe the version refers to dbt's schema format rather than the data's schema version. 
3) We need to specify schema names for the models. You can do that in dbt_project.yml. 

I added this: 

    address_lists:
      optimism:
        +schema: optimism_airdrop1
        +materialized: view

which means your view will end up being named `optimism_airdrop1.address_lists` 

and 
      optimism:
        +schema: tokens_optimism
        +materialized: view

which means your view will end up being names `tokens_optimism.erc20`. Otherwise, it would have conflicted with erc20.tokens. 

ps I promise there's going to be a way for you to run dbt yourself coming soon. We know this workflow needs to be better!

**Comment by dot2dotseurat** [comment]
@MSilb7 op_airdrop1_address_list is failing on a syntax error but I'm pretty sure it's actually due to the size of the file not an actual syntax issue.

Can I remove op_airdrop1_address_list from the pull request? Then we can merge the rest of it and I will work on a solution for uploading large lists of values.

**Comment by MSilb7** [comment]
> @MSilb7 there's a couple changes needed, I'm going to add them over here so I can use our dbt cloud tests #1306
> 
> 1. There were some missing quotes in the optimism token addresses files.
> 2. Schema files need to be be verison 2. It's misleading but I believe the version refers to dbt's schema format rather than the data's schema version.
> 3. We need to specify schema names for the models. You can do that in dbt_project.yml.
> 
> I added this:
> 
> ```
> address_lists:
>   optimism:
>     +schema: optimism_airdrop1
>     +materialized: view
> ```
> 
> which means your view will end up being named `optimism_airdrop1.address_lists`
> 
> and optimism: +schema: tokens_optimism +materialized: view
> 
> which means your view will end up being names `tokens_optimism.erc20`. Otherwise, it would have conflicted with erc20.tokens.
> 
> ps I promise there's going to be a way for you to run dbt yourself coming soon. We know this workflow needs to be better!

Got it, no problem. Learning it all as I go haha, I've never used dbt before so I have zero expectation for what makes sense.

On version - good callout. I thought of it like "table version" rather than "database version." I'll check out your changes. I can drop the airdrop list from this PR.

I haven't had a chance yet to find the symbol & erc types yet for the NFT table (last outstanding thing), but should be able to this weekend or later this week!

**Comment by jeff-dude** [comment]
@dot2dotseurat should we close this PR now that #1306 was merged?

then @MSilb7 can merge master into his forked repo and begin to add enhancement to nft schema & file when ready?

**Comment by dot2dotseurat** [comment]
> @dot2dotseurat should we close this PR now that #1306 was merged?
> 
> then @MSilb7 can merge master into his forked repo and begin to add enhancement to nft schema & file when ready?

Yes, these changes have been merged so we can close this. (commits are still attributed to @MSilb7, it was just easier to run the tests from the internal pr). 

@MSilb7, awesome work. killing it learning dbt on hard mode.

**Pull request closed without merging**